scripts/webpack/ConfigurationPlugin.ts

```diff
@@ -42,32 +68,85 @@ export class ConfigurationPlugin implements webpack.WebpackPluginInstance {
         name: 'ConfigurationPlugin',
         stage: webpack.Compilation.PROCESS_ASSETS_STAGE_OPTIMIZE_INLINE,
       },
-      async (assets) => {
-        const asset = assets[this.options.filename] as
-          | webpack.sources.RawSource
-          | undefined;
+      this.handleTemplateIndex.bind(this, compilation)
+    );
+  }
+
+  protected async handleTemplateIndex(
+    compilation: webpack.Compilation
+  ): Promise<void> {
+    const logger = compilation.getLogger('ConfigurationPlugin');
+
+    logger.info('Looking for configuration template');
+    const asset = compilation.assets[this.options.filename] as
+      | webpack.sources.RawSource
+      | undefined;
 
-        if (!asset) return Promise.resolve();
+    if (!asset) return Promise.resolve();
 
-        const contents = asset.buffer().toString();
+    const contents = asset.buffer().toString();
 
-        let values = await getConfigurationValues();
+    logger.info('Templating configuration file');
+    // Template the file with the configuration values.
+    const transformedContents = await templateIndex(
+      contents,
+      this.configurationValues
+    );
+
+    logger.info('Emitting configuration file asset');
+    // Create the new templated file.
+    const modifiedSource = new webpack.sources.RawSource(
+      transformedContents,
+      false
+    );
 
-        // Apply configuration value overrides, if any.
-        if (this.options.overrides) {
-          values = Object.assign({}, values, this.options.overrides);
-        }
+    compilation.emitAsset(this.options.outputFilename, modifiedSource);
+  }
 
-        // Template the file with the configuration values.
-        const transformedContents = await templateIndex(contents, values);
+  protected async handlePrepareEnvironment(logger: Logger): Promise<void> {
+    let config: string | undefined;
 
-        // Create the new templated file.
-        const modifiedSource = new webpack.sources.RawSource(
-          transformedContents,
-          false
-        );
-        compilation.emitAsset(this.options.outputFilename, modifiedSource);
-      }
+    const providedKubectlContext = process.env.hasOwnProperty(
+      'HAPPA_KUBECTL_CONTEXT'
     );
+
+    if (providedKubectlContext) {
```

Comment:
If we provided a context, then we fetch the `ConfigMap` from the installation's management cluster. Otherwise, we don't 😬